app/controllers/books_controller.rb

```diff
@@ -91,6 +91,6 @@ def set_book
   end
 
   def check_role
-    redirect_to books_url unless @book.user_id == current_user.id
+    redirect_to books_url unless @book.user_id == current_user.id || current_user.admin?
```

Comment:
真偽値なので`?`は不要かなと思いました。
```suggestion
    redirect_to books_url unless @book.user_id == current_user.id || current_user.admin
```